src/index.ts

```diff
@@ -0,0 +1,477 @@
+   */
+  api: API;
+  /**
+   * Is Quote Tool read-only
+   */
+  readOnly: boolean;
+  /**
+   * Placeholder for Quote Tool
+   */
+  private _quotePlaceholder: string;
+  /**
+   * Caption placeholder for Quote Tool
+   */
+  private _captionPlaceholder: string;
+  /**
+   * Quote's data
+   */
+  private _data: QuoteData;
+  /**
+   * Quote Tool's CSS classes
+   */
+  private _CSS: QuoteCSS;
+
+  /**
+   * Render plugin`s main Element and fill it with saved data
+   *
+   * @param {object} params - constructor params
+   * @param {QuoteData} params.data - previously saved data
+   * @param {QuoteConfig} params.config - user config for Tool
+   * @param {API} params.api - editor.js api
+   * @param {boolean} params.readOnly - read only mode flag
+   */
+  constructor({ data, config, api, readOnly }: QuoteParams) {
+    const { ALIGNMENTS, DEFAULT_ALIGNMENT } = Quote;
+
+    this.api = api;
+    this.readOnly = readOnly;
+
+    this._quotePlaceholder =
+      config.quotePlaceholder || Quote.DEFAULT_QUOTE_PLACEHOLDER;
+    this._captionPlaceholder =
+      config.captionPlaceholder || Quote.DEFAULT_CAPTION_PLACEHOLDER;
+
+    this._data = {
+      text: data.text || "",
+      caption: data.caption || "",
+      alignment:
+        (Object.values(ALIGNMENTS).includes(data.alignment) &&
+          data.alignment) ||
+        config.defaultAlignment ||
+        DEFAULT_ALIGNMENT,
+    };
+    this._CSS = {
+      baseClass: this.api.styles.block,
+      wrapper: "cdx-quote",
+      text: "cdx-quote__text",
+      input: this.api.styles.input,
+      caption: "cdx-quote__caption",
+    };
+  }
+
+  /**
+   * Notify core that read-only mode is supported
+   *
+   * @returns {boolean}
+   */
+  static get isReadOnlySupported(): boolean {
+    return true;
+  }
+
+  /**
+   * Get Tool toolbox settings
+   * icon - Tool icon's SVG
+   * title - title to show in toolbox
+   *
+   * @returns {{icon: string, title: string}}
+   */
+  static get toolbox(): { icon: string; title: "Quote" } {
+    return {
+      icon: IconQuote,
+      title: "Quote",
+    };
+  }
+
+  /**
+   * Empty Quote is not empty Block
+   *
+   * @public
+   * @returns {boolean}
+   */
+  static get contentless(): boolean {
+    return true;
+  }
+
+  /**
+   * Allow to press Enter inside the Quote
+   *
+   * @public
+   * @returns {boolean}
+   */
+  static get enableLineBreaks(): boolean {
+    return true;
+  }
+
+  /**
+   * Default placeholder for quote text
+   *
+   * @public
+   * @returns {string}
+   */
+  static get DEFAULT_QUOTE_PLACEHOLDER(): string {
+    return "Enter a quote";
+  }
+
+  /**
+   * Default placeholder for quote caption
+   *
+   * @public
+   * @returns {string}
+   */
+  static get DEFAULT_CAPTION_PLACEHOLDER(): string {
+    return "Enter a caption";
+  }
+
+  /**
+   * Allowed quote alignments
+   *
+   * @public
+   * @returns {{left: string, center: string}}
+   */
+  static get ALIGNMENTS(): { left: "left"; center: "center" } {
```

Comment:
I'd suggest to create Enum `Alignment` with `Left` and `Center`

Reply:
done👍🏼